tmt/utils.py

```diff
@@ -6183,3 +6183,29 @@ def __exit__(self, *args: Any) -> None:
     @property
     def duration(self) -> datetime.timedelta:
         return self.end_time - self.start_time
+
+
+def retry(
+        function_to_retry: Callable[..., Any],
+        retry_count: int,
+        delay: int,
+        error_message: str,
+        *args: Optional[Any],
+        **kwargs: Optional[Any]
+        ) -> Any:
+    """ Retry functionality to be used elsewhere in the code.
+
+    :param function_to_retry: function to be called with any amount
+    of arguments of Any type, returning result of Any type
+    :param retry_count: number of tries to call the function
+    :param delay: amount of seconds to wait before a new try
+    :param error_message: message explaining retry reason
+    """
+
+    for _ in range(retry_count):
+        try:
+            function_to_retry(*args, **kwargs)
+            break
+        except Exception:
+            print(f"{error_message}")
```

Comment:
Please, pass `logger` to `retry`, and use it for logging instead of bare `print()`. Check out `tmt.utils.create_directory()`, for example.

Reply:
Plus, we do need to log the important info: an attempt failed, how many are left, and when we would try again. Check out https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L5292C1-L5297C1, for example. Similarly, `retry` should log something like this:

```python
for i in range(retries):
    ...

    except Exception as exc:
        logger.debug(
            'retry',
            f"{label} failed, {retries - i} retries left, trying again in {delay:.2f} seconds")
        time.sleep(delay)
```